### ISSUE TITLE
Actual extension ignored for names with default extension

### ISSUE THREAD

**Issue by Dunedan** [chriskuehl/identify#21]
All names with a default extension in `extensions.py` (https://github.com/chriskuehl/identify/blob/4b664c4c7268cb650650a58edc01980e036064ea/identify/extensions.py#L129-L141) aren't properly detected if they have a file extension. Instead the actual file extension is simply ignored:
```
foo@bar:~$ identify-cli README  # works
["file", "non-executable", "plain-text", "text"]
foo@bar:~$ identify-cli README.md  # doesn't work ("plain-text" should be "markdown")
["file", "non-executable", "plain-text", "text"]
foo@bar:~$ identify-cli README2.md  # works
["file", "markdown", "non-executable", "text"]
```

**Comment by asottile** [comment]
Indeed!  I'll get a patch for this going :)

**Comment by chriskuehl** [comment]
Nice find, thanks for reporting it!

**Comment by asottile** [comment]
This is available in 1.0.4!
